snuba/clickhouse/astquery.py

```diff
@@ -0,0 +1,150 @@
+from typing import Optional
+
+from snuba import settings
+from snuba.clickhouse.query import ClickhouseQuery
+from snuba.clickhouse.formatter import ClickhouseExpressionFormatter
+from snuba.query.parsing import ParsingContext
+from snuba.query.query import Query
+from snuba.request.request_settings import RequestSettings
+
+
+class AstClickhouseQuery(ClickhouseQuery):
```

Comment:
Note for later: it'd be nice after we've fully switched over to put this back in `snuba.clickhouse.query` and rename this as `ClickhouseQuery` (removing the existing `ClickhouseQuery`) since at that point the AST distinction should be overly specific.

Reply:
Yeah, that's the plan